test/conformance/cts_exe.py

```diff
@@ -10,25 +10,44 @@
 # Printing conformance test output from gtest and checking failed tests with match files.
 # The match files contain tests that are expected to fail.
 
+import os
+import shlex
 import sys
 from argparse import ArgumentParser
 import subprocess  # nosec B404
 import signal
 import re
 from collections import OrderedDict
 
-if __name__ == '__main__':
+
+def _print_cmdline(cmd_args, env, cwd, file=sys.stderr):
+    cwd = shlex.quote(cwd)
+    env_args = " ".join(
+        "%s=%s" % (shlex.quote(k), shlex.quote(v)) for k, v in env.items()
+    )
+    cmd_str = " ".join(map(shlex.quote, cmd_args))
+    print(f"### env -C {cwd} -i {env_args} {cmd_str}", file=file)
+
+
+if __name__ == "__main__":
 
     parser = ArgumentParser()
     parser.add_argument("--test_command", help="Ctest test case")
     parser.add_argument("--devices_count", type=str, help="Number of devices on which tests will be run")
     parser.add_argument("--platforms_count", type=str, help="Number of platforms on which tests will be run")
     args = parser.parse_args()
+    invocation = [
+        args.test_command,
+        "--gtest_brief=1",
+        f"--devices_count={args.devices_count}",
+        f"--platforms_count={args.platforms_count}",
+    ]
+    _print_cmdline(invocation, os.environ, os.getcwd())
 
-    result = subprocess.Popen([args.test_command, '--gtest_brief=1',        # nosec B603
```

Comment:
> > That bandit failure looks like nonsense.
> 
> Right. So there are two:
> 
>     * [b602_subprocess_popen_with_shell_equals_true](https://bandit.readthedocs.io/en/latest/plugins/b602_subprocess_popen_with_shell_equals_true.html)
> 
>     * [b603_subprocess_without_shell_equals_true](https://bandit.readthedocs.io/en/latest/plugins/b603_subprocess_without_shell_equals_true.html)
> 
> 
> So basically they don't want us calling subprocesses?
> 
> This seems like an unwanted lint in testing infrastructure which relies on executing subprocesses. Can we disable this?

You've deleted the `# nosec B603` comment, I assume this is doing what you ask. As for B602, looks wrong to me. Since `shell=False` is the default, why not just delete it?

Reply:
Thanks for the hint. Didn't realise that was related. Seems obvious in retrospect